File: CPP/Hardware.cpp
class Hardware : public Product
{
	// private attributes
	private:
		string brand;
		string model;

	// public attributes	
	public:
		// constructor with parameter
		Hardware(string brand, string model)
		{
			this->brand = brand;
			this->model = model;
		}
		// constructor
		Hardware(){
			this->brand = "";
			this->model = "";
		}

		// method for brand attribute
		void setBrand(string brand)
		{
			this->brand = brand;
		}
		string getBrand()
		{
			return this->brand;
		}

		// method for model attribute
		void setModel(string model)
		{
			this->model = model;
		}
		string getModel()
		{
			return this->model;
		}

		// destructor
		~Hardware()
		{

		}

};

File: PHP/Hardware.php
<?php 

    // class for Hardware
    class Hardware extends Product
    {
        // public attributes
        public $brand;
        public $model;

        // constructor
        public function __construct($brand = "", $model = "")
        {
            $this->brand = $brand;
            $this->model = $model;            
        }   

        // methods for brand attribute
        public function setBrand($brand)
        {
            $this->brand = $brand;
        }   
        public function getBrand()
        {
            return $this->brand;
        }

        // methods for model attribute
        public function setModel($model)
        {
            $this->model = $model;
        }   
        public function getModel()
        {
            return $this->model;
        }

        // destructor
        public function __destruct()
        {
            
        }
    }
?>